hawc/apps/epiv2/models.py

```diff
@@ -0,0 +1,322 @@
+import reversion
+from django.db import models
+
+from ..study.models import Study
+from . import constants, managers
+
+
+# TODO: set all max_lengths based on the needs of each field
+class AgeProfile(models.Model):
+    objects = managers.AgeProfileManger()
+
+    name = models.CharField(unique=True, max_length=128,)
+
+    def __str__(self):
+        return self.name
+
+
+class Country(models.Model):
+    objects = managers.CountryManager()
+
+    code = models.CharField(blank=True, max_length=2)
+    name = models.CharField(unique=True, max_length=64)
+
+    class Meta:
+        ordering = ("name",)
+        verbose_name_plural = "Countries"
+
+    def __str__(self):
+        return self.name
+
+
+class MeasurementType(models.Model):
+    objects = managers.MeasurementTypeManager()
+
+    description = models.CharField(max_length=128,)
+
+    def __str__(self):
+        return self.description
+
+
+class StudyPopulation(models.Model):
+    objects = managers.StudyPopulationManager()
+
+    study = models.ForeignKey(Study, on_delete=models.CASCADE, related_name="study_populations_v2")
+    study_design = models.CharField(
+        max_length=128, choices=constants.StudyDesign.choices, blank=True, null=True,
+    )
+    source = models.CharField(
+        max_length=128, choices=constants.Source.choices, blank=True, null=True,
+    )
+    age_profile = models.ManyToManyField(
+        AgeProfile,
+        blank=True,
+        help_text='Select all that apply. Note: do not select "Pregnant women" if pregnant women '
+        + "are only included as part of a general population sample",
+        verbose_name="Population age category",
+    )
+    age_description = models.CharField(
+        max_length=64, blank=True, null=True, verbose_name="Population age details",
+    )
+    sex = models.CharField(default="U", max_length=1, choices=constants.Sex.choices,)
+    race = models.CharField(
+        max_length=128, blank=True, null=True, verbose_name="Population race/ethnicity"
+    )
+    summary = models.CharField(
+        max_length=128,
+        verbose_name="Population Summary",
+        help_text="Breifly describe the study population (e.g., Women undergoing fertility treatment).",
+    )
+    study_name = models.CharField(
+        max_length=64,
+        blank=True,
+        null=True,
+        verbose_name="Study name (if applicable",
+        help_text="Typically available for cohorts. Abbreviations provided in the paper are fine",
+    )
+    countries = models.ManyToManyField(Country, blank=True,)
+    region = models.CharField(
+        max_length=128, blank=True, null=True, verbose_name="Other geographic information"
+    )
+    # TODO: is there a better way of doing this with datetime objects to make it searchable
```

Comment:
no, but we could have some custom regex validators?  Not sure we can define it that tight

https://docs.djangoproject.com/en/4.0/ref/validators/#regexvalidator